### PR TITLE
MTE Inventory updates

### PR DESCRIPTION
**Pull request by mitchej123**
MTE Inventory & Fluid
* Separate Input/Output inventory
* Use a LinkedHashMap to ensure inventory orders are deterministic
* Input/Output work on either Input/Output inventories
* GUIs on MTE hatches
* More dyanmic packet with packetFeatures
* Add IMTE_HasModes for MultiBlockPart
* MTE fluid inventory
* working controller GUI
* locked inventory selection work
* input and output locking of inventories

Ticking Covers
* Parts now register covers with the controller
* Controllers now tick covers on parts
* Break cover ticking out into `tickCoverAtSide`

CoverInfo refactor

### CODE REVIEW

File: src/main/java/gregtech/loaders/preload/GT_Loader_MultiTileEntities.java [hunk 1]
```diff
@@ -16,22 +19,23 @@ public void run() {
...
-        //            .register();
-        //
-        //        aRegistry.create(18000, MultiBlockPart.class)
-        //            .name("Test Casing")
-        //            .category("Multiblock Casing")
-        //            .setBlock(aMachine)
-        //            .material(Materials.Cobalt)
-        //            .texture("metalwall")
-        //            .register();
+        aRegistry
```
Comment (mitchej123) [inline src/main/java/gregtech/loaders/preload/GT_Loader_MultiTileEntities.java:22]: TODO: Comment this out before merging

**Comment by mitchej123** [comment]
PR is ready for review.  The MTE changes should have minimal to no impact on anything as they're not yet in use, and the registration is disabled.

The CoverInfo refactor will impact existing words, but has been tested on the full pack and *should* be backwards compatible.

**Comment by Dream-Master** [comment]
Machines not work anymore it seems to freez the state when the machine was saved before this patch
Cables have no connecting graphic. If it hit with Wirecutter is show only a messages but no animation

**Comment by mitchej123** [comment]
Will dig into, seems like subtle bug in one of the `letsEnergyIn` or `letsEnergyOut` refactors.

**Comment by Dream-Master** [comment]
I cant find another bug. Works in my sp game

**Review by Dream-Master — APPROVED** [review]

**Comment by Dream-Master** [comment]
I was wrong it not work fully. @mitchej123 we need some more fixes so that gui covers work again

**Pull request merged**